### ISSUE TITLE
MVVM.Toolkit: WeakReferenceMessenger.Send throws System.IO.FileLoadException

### ISSUE THREAD

**Issue by DhananjayDhamale** [CommunityToolkit/dotnet#86]
### Describe the bug

Hi All, 

We are migrating our class library application from MVVMLight to MVVMToolkit. We are using latest nuget package (V 7.1.2)  We are using WeakReferenceMessenger to send message.

WeakReferenceMessenger.Default.Send(new XXXLoaded(this)); The send call throws "System.IO.FileLoadException" as below - 

_System.IO.FileLoadException: 'Could not load file or assembly 'System.Buffers, Version=4.0.2.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)'_

I have analyzed more and found that Microsoft.Toolkit.Mvvm refers System.Buffer with ver 4:0:2:0 and it also refers System.Memory which internally refers System.Buffer with ver 4:0:3:0. As nuget package installs latest version of System.Buffer (due to compatibility), application throws runtime exception.

Note: I cannot use .config file to redirect versions.
![MVVMToolKit-MANIFEST](https://user-images.githubusercontent.com/83224901/149062591-f2410015-350e-4e77-94a2-dd68db634bac.jpg)
![System Memory_Manifest](https://user-images.githubusercontent.com/83224901/149062610-b71d9117-4270-459e-9a3c-85b8187a285d.jpg)

### Regression

7.1.2

### Reproducible in sample app?

- [ ] This bug can be reproduced in the sample app.

### Steps to reproduce

```text
Did not try in Sample app.
```

### Expected behavior

WeakReferenceMessenger.Default.Send should not throw exception. 

### Windows Build Number

- [ ] Windows 10 1809 (Build 17763)
- [ ] Windows 10 1903 (Build 18362)
- [X] Windows 10 1909 (Build 18363)
- [ ] Windows 10 2004 (Build 19041)
- [ ] Windows 10 20H2 (Build 19042)
- [ ] Windows 10 21H1 (Build 19043)
- [ ] Windows 11 21H2 (Build 22000)
- [ ] Other (specify)

### App minimum and target SDK version

- [ ] Windows 10, version 1809 (Build 17763)
- [ ] Windows 10, version 1903 (Build 18362)
- [ ] Windows 10, version 1909 (Build 18363)
- [ ] Windows 10, version 2004 (Build 19041)
- [ ] Other (specify)

### Visual Studio Version

2017

### Visual Studio Build Number

15.9.13

### Nuget packages

 Microsoft.Toolkit.Mvvm Version 7.1.2

### Help us help you

Yes, but only if others can assist.

**Comment by Sergio0694** [comment]
Hi @DhananjayDhamale, what runtime are you targeting?
Are you on .NET Framework by any chance?

**Comment by DhananjayDhamale** [comment]
> Hi @DhananjayDhamale, what runtime are you targeting? Are you on .NET Framework by any chance?

Hi Sergio0694, 
I am using .NET Framework 4.8.

**Comment by Sergio0694** [comment]
Ah, I see. This is an issue with how .NET Framework resolves assembly binding redirects when referencing packages that have indirect references to any of the OOB libraries from the BCL (eg. `System.Memory` in this case). It's not something we can address from the Toolkit, as we're just targeting .NET Standard 2.0 in this case and pulling those dependencies in, which are also for .NET Standard. You will have to address this on your end by manually configuring binding redirects for this assembly 🥲

**Comment by efdiloreto** [comment]
Does anyone solve this problem? I've tried to manually doing the binding redirect without any success.
```
    <configuration>
      <runtime>
        <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
          <dependentAssembly>
            <assemblyIdentity name="System.Buffers" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
            <bindingRedirect oldVersion="4.0.2.0" newVersion="4.5.1.0" />
          </dependentAssembly>
        </assemblyBinding>
      </runtime>
    </configuration>
```

**Comment by view21cy** [comment]
i also have this probelm.
would you please show the code how to modify this?
"manually configuring binding redirects for this assembly”？？

**Comment by whit-tj** [comment]
Anyone find out how to handle this? I can't get any binding redirects to resolve this issue. Uninstalling and reinstalling but the binding redirects seem to have 0 effect. Is there any combo of dependency versions that work? 7.0.2 worked but we need the bug fixes in 7.1.2,
